src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -259,21 +259,18 @@ public static bool IsKeyValuePair(this Type type)
                 }
             }
 
-            // For correctness, throw if multiple ctors have [JsonConstructor], even if one or more are non-public.
-            ConstructorInfo? dummyCtorWithAttribute = ctorWithAttribute;
-
-            constructors = type.GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance);
-            foreach (ConstructorInfo constructor in constructors)
+            // Search for non-public ctors with [JsonConstructor].
+            foreach (ConstructorInfo constructor in type.GetConstructors(BindingFlags.NonPublic | BindingFlags.Instance))
```

Comment:
do we actually need two loops now?

Reply:
the non-public one has slightly different logic (i.e. it doesn't single out parameterless ctors)